SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp

```diff
@@ -1363,6 +1365,138 @@ bool RunSchemaCompiler()
 	}
 }
 
+#define SAFE_TRYGET(Value, Type, OutParam)                                                                                                 \
+	do                                                                                                                                     \
+	{                                                                                                                                      \
+		if (!Value->TryGet##Type(OutParam))                                                                                                \
+		{                                                                                                                                  \
+			UE_LOG(LogSpatialGDKSchemaGenerator, Error, TEXT("Failed to get %s as type %s"), TEXT(#Value), TEXT(#Type));                   \
+			return false;                                                                                                                  \
+		}                                                                                                                                  \
+	} while (false)
+
+#define SAFE_TRYGETFIELD(Value, Type, FieldName, OutParam)                                                                                 \
+	do                                                                                                                                     \
+	{                                                                                                                                      \
+		if (!Value->TryGet##Type##Field(TEXT(FieldName), OutParam))                                                                        \
+		{                                                                                                                                  \
+			UE_LOG(LogSpatialGDKSchemaGenerator, Error, TEXT("Failed to get field %s of type %s from %s"), TEXT(FieldName), TEXT(#Type),   \
+				   TEXT(#Value));                                                                                                          \
+			return false;                                                                                                                  \
+		}                                                                                                                                  \
+	} while (false)
+
+bool ExtractComponentSetFromSchemaJson(const FString& SchemaJsonPath, TMap<uint32, FComponentIDs>& OutComponentSetMap)
+{
+	TSharedPtr<FJsonValue> SchemaBundleJson;
+	{
+		TUniquePtr<FArchive> SchemaFile(IFileManager::Get().CreateFileReader(*SchemaJsonPath));
+		if (!SchemaFile)
+		{
+			return false;
```

Comment:
should probably log a warning/error here- unless we expect this path to typically be incorrect?